iFixit/react-commerce
# Upgrade strapi v4.15.0

closes https://github.com/iFixit/ifixit/issues/50668

Oh boy, this one took quite a bit to get it right. Strapi has made some Typescript-related breaking changes not trivial to figure out.
This version should fix the problem described by https://github.com/iFixit/ifixit/issues/50668 while still being compatible with the publisher plugin (which breaks the build on the latest version).

## QA

1. Open [Strapi govinor preview](https://upgrade-strapi-v4-15-0.govinor.com/admin/content-manager/collectionType/api::reusable-section.reusable-section/3)
2. Go to reusable section content type, and visit one of the existing entries
3. Under "placement", try to select the "All Tools" product list and verify that you're able to select the product list
